includes/Builders/Prompt_Builder_With_WP_Error.php

```diff
@@ -0,0 +1,163 @@
+ * @method string|WP_Error generate_text() Generates text from the prompt.
+ * @method list<string>|WP_Error generate_texts(?int $candidateCount = null) Generates multiple text candidates from the prompt.
+ * @method File|WP_Error generate_image() Generates an image from the prompt.
+ * @method list<File>|WP_Error generate_images(?int $candidateCount = null) Generates multiple images from the prompt.
+ * @method File|WP_Error convert_text_to_speech() Converts text to speech.
+ * @method list<File>|WP_Error convert_text_to_speeches(?int $candidateCount = null) Converts text to multiple speech outputs.
+ * @method File|WP_Error generate_speech() Generates speech from the prompt.
+ * @method list<File>|WP_Error generate_speeches(?int $candidateCount = null) Generates multiple speech outputs from the prompt.
+ */
+class Prompt_Builder_With_WP_Error extends Prompt_Builder {
+
+	/**
+	 * WordPress error instance, if any error occurred during method calls.
+	 *
+	 * @since n.e.x.t
+	 * @var WP_Error|null
+	 */
+	private ?WP_Error $error = null;
+
+	/**
+	 * List of methods that terminate the fluent interface and return a result.
+	 *
+	 * Technically a map, simply for faster lookups.
+	 *
+	 * @since n.e.x.t
+	 * @var array<string, bool>
+	 */
+	private array $terminate_methods = array(
+		'generate_result'               => true,
+		'generate_text_result'          => true,
+		'generate_image_result'         => true,
+		'generate_speech_result'        => true,
+		'convert_text_to_speech_result' => true,
+		'generate_text'                 => true,
+		'generate_texts'                => true,
+		'generate_image'                => true,
+		'generate_images'               => true,
+		'convert_text_to_speech'        => true,
+		'convert_text_to_speeches'      => true,
+		'generate_speech'               => true,
+		'generate_speeches'             => true,
+	);
+
+	/**
+	 * Magic method to proxy snake_case method calls to their PHP AI Client camelCase counterparts.
+	 *
+	 * This allows WordPress developers to use snake_case naming conventions. It also catches any exceptions thrown,
+	 * stores them, and returns a WP_Error when a terminate method is called.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string            $name      The method name in snake_case.
+	 * @param array<int, mixed> $arguments The method arguments.
+	 * @return mixed The result of the parent method call.
+	 */
+	public function __call( string $name, array $arguments ) {
+		// This may throw, which is fine because calls to methods that don't exist always throw.
+		$callable = $this->get_builder_callable( $name );
+
+		/*
+		 * If an error occurred in a previous method call, either return the error for terminate methods,
+		 * or return the same instance for other methods to maintain the fluent interface.
+		 */
+		if ( null !== $this->error ) {
+			if ( isset( $this->terminate_methods[ $name ] ) ) {
```

Comment:
Since this is referenced twice, let's just make this a tiny, declarative `is_terminating_method( $name )` method.

Also, we can make `$terminate_mtehods` a static property. Same with the method I'm proposing.